components/quests/quest.tsx

```diff
@@ -33,20 +33,41 @@ const Quest: FunctionComponent<QuestProps> = ({
   id,
   expired,
 }) => {
-  const { completedQuestIds } = useContext(QuestsContext);
+  const { completedQuestIds, boostedQuests } = useContext(QuestsContext);
   const isCompleted = useMemo(
     () => completedQuestIds.includes(id),
     [id, completedQuestIds]
   );
-  const [boostedQuests, setBoostedQuests] = useState<number[]>([]);
+  const [boost, setBoost] = useState<Boost>();
+  const [isQuestBoosted, setIsQuestBoosted] = useState<boolean>(false);
 
-  const fetchBoostedQuests = useCallback(async () => {
-    const response = await getBoostedQuests();
-    setBoostedQuests(response);
+  const checkIfBoostedQuest = useCallback(async () => {
+    if (!boostedQuests) return;
+    if (boostedQuests.length > 0 && boostedQuests.includes(id))
+      setIsQuestBoosted(true);
   }, []);
 
+  const fetchBoosts = useCallback(async (id: string) => {
+    try {
+      const response = await getBoosts();
+      if (!response) return;
+      const boost = response.find((b: Boost) =>
+        b.quests.includes(parseInt(id))
+      );
+      if (!boost) return;
+      setBoost(boost);
+    } catch (err) {
+      console.log("Error while fetching boost by id", err);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!isQuestBoosted) return;
+    fetchBoosts(id.toString());
+  }, [isQuestBoosted]);
```

Comment:
Shouldn't we add "id" as a dependency: `  }, [isQuestBoosted, id]);` ? In case we want to dynamically change a card (this is not useful at the moment, but it would avoid weird behaviours in futur use cases I think), otherwise the "1000 USDC" tag would stay, even if it's not the same value as the former card or not boosted at all. What do you think?

Reply:
yes. this is a fair feedback. Adding props which are involved in the operation is a good practice. Thank you for pointing this out!